### ISSUE TITLE
Still recommend redundant `gAMA`?

### ISSUE THREAD

**Issue by svgeesus** [w3c/PNG-spec#151]
From [4.3 Colour spaces](https://w3c.github.io/PNG-spec/#4Concepts.ColourSpaces):

> It is recommended that explicit gamma information also be provided when either the first, second or third method is used, for use by PNG decoders that do not support CICP metadata, full ICC profiles or the sRGB colour space. 

Two reasons to not recommend  redundant `gAMA` when a better method is used:

1. Only a simple power function, not piecewise, so provides only an approximation of sRGB, PQ, HLG, Adobe RGB and ProPhoto transfer functions, for example
2. Spec says to ignore it, either sometimes or always. See:
  2.1. [cICP](https://w3c.github.io/PNG-spec/#11cICP) 

  > When the cICP chunk is present, decoders that recognize it SHALL ignore the following chunks:
    iCCP
    gAMA
    cHRM
    sRGB

  2.2 [iCCP]()

  > When the iCCP chunk is present, PNG decoders that recognize it and are capable of colour management shall ignore the [gAMA](https://w3c.github.io/PNG-spec/#gama-image-gamma) and [cHRM](https://w3c.github.io/PNG-spec/#chrm-primary-chromaticities-and-white-point) chunks and use the iCCP chunk instead and interpret it according to [[ICC](https://w3c.github.io/PNG-spec/#bib-icc)]. PNG decoders that are used in an environment that is incapable of full-fledged colour management should use the [gAMA](https://w3c.github.io/PNG-spec/#gama-image-gamma) and [cHRM](https://w3c.github.io/PNG-spec/#chrm-primary-chromaticitieis-and-white-point) chunks if present.

  2.3 [sRGB]()

  > When the sRGB chunk is present, it is recommended that decoders that recognize it and are capable of colour management ignore the [gAMA](https://w3c.github.io/PNG-spec/#gama-image-gamma) and [cHRM](https://w3c.github.io/PNG-spec/#chrm-primary-chromaticities-and-white-point) chunks and use the sRGB chunk instead. Decoders that recognize the sRGB chunk but are not capable of colour management are recommended to ignore the [gAMA](https://w3c.github.io/PNG-spec/#gama-image-gamma) and [cHRM](https://w3c.github.io/PNG-spec/#chrm-primary-chromaticities-and-white-point) chunks, and use the values given above as if they had appeared in [gAMA](https://w3c.github.io/PNG-spec/#gama-image-gamma) and [cHRM](https://w3c.github.io/PNG-spec/#chrm-primary-chromaticities-and-white-point) chunks.

  In other words, the [value for `gAMA` when used with `sRGB` is fixed](https://w3c.github.io/PNG-spec/#table-gama-and-chrm-values-for-srgb), could be inferred, and so the explicit `gAMA` adds no value but could be incorrect.

I feel that `gAMA` (and `cHRM`) are legacy chunks; they should still be supported but we should revisit recommending adding them when better info is available in other chunks.

**Comment by ProgramMax** [comment]
I strongly agree.

Today, the gAMA and cHRM chunks are for extremely legacy systems. Recommending those chunks be included as a fallback is only useful during a transitory phase. We're well past that transitory phase now. So the only value is for those extremely legacy systems which are not aware of sRGB or iCCP chunks.

I do not know of any highly used decoders which are not sRGB-aware.
I'm sure there are some out there with very limited use. But do we want to **recommend** all images contain extra bytes for these rare, limited systems?

I am sure there are lots of decoders which simply ignore color space information entirely. I bet something like 99% of projects which use lodepng just drop the library in and decode the raw pixel values and do not account for color space. In those projects, all of the color space chunks are unused anyway. So we can ignore those use cases when talking about color correctness.

Anyway, the value provided during the transitory phase has passed. I think we should no longer recommend gAMA & cHRM fallbacks.

**Comment by FooIbar** [comment]
The gAMA and cHRM chunks are still recommend in:

[11.3.2.3 iCCP Embedded ICC profile](https://w3c.github.io/png/#11iCCP)
> A PNG encoder that writes the iCCP chunk is encouraged to also write gAMA and cHRM chunks that approximate the ICC profile, to provide compatibility with applications that do not use the iCCP chunk.

[11.3.2.5 sRGB Standard RGB color space](https://w3c.github.io/png/#srgb-standard-colour-space)
> It is recommended that a PNG encoder that writes the sRGB chunk also write a gAMA chunk (and optionally a cHRM chunk) for compatibility with decoders that do not use the sRGB chunk.

Should those be removed too?

**Comment by ProgramMax** [comment]
Yes, those should be reworded to something like "when supporting legacy systems that are unaware of sRGB or iCCP chunks (rare today), ..."

**Comment by svgeesus** [comment]
It seems we have consensus on this, so I will make a PR.
